presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -721,16 +739,21 @@ public synchronized void finishInsertIntoExistingPartition(
         }
     }
 
-    private String getPartitionName(String databaseName, String tableName, List<String> partitionValues)
+    private String getPartitionName(Table table, List<String> partitionValues)
     {
-        Table table = getTable(databaseName, tableName)
-                .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
         List<String> columnNames = table.getPartitionColumns().stream()
                 .map(Column::getName)
                 .collect(toImmutableList());
         return makePartName(columnNames, partitionValues);
     }
 
+    private String getPartitionName(String databaseName, String tableName, List<String> partitionValues)
```

Comment:
Nit: move this method above, as you usually chain methods down rather than up (the root method is at the bottom of the file)